mtp_api/apps/security/views.py

```diff
@@ -528,6 +528,7 @@ class Meta:
 class CheckView(
     mixins.ListModelMixin,
     mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
```

Comment:
this enforces no restrictions on which user is set on `assigned_to` or what fields can be edited. this potentially leaves the api open to incorrect changes. e.g. if the status field is changed (by accident or whatever), then the proper business logic doesn't kick in.

do you think it might be worth having an "action" (like accept or reject below) that automatically assigns the user from the request? and (if need be) another to clear such assignment?

Reply:
I mean, we could have seperate endpoints, but it seems more RESTful to have the `PATCH` method use the same url as the `GET`. What we could do though is stop `status` being mutated within the generic `update` method being used by the `CheckSerializer`? And for other fields that we don't want to mutate in any use case we could set them as `read_only` within the serializer?

Validation of the request against the user is another thing that could be done in the serializer (we've just pushed a commit that wasn't there when you reviewed that prevents assignment when someone else is assigned), just so we're encapsulating all the validation functionality all in one place. I'm tempted to leave out the validation of the request user, mainly because it would mean that if we ever made a page where the head of FIU could assign cases to her subordinates it would mean we could implement it without changes to the API repo. What do you think?

Reply:
i agree this is more restful. yeah i think patch validation or read-only fields are a good idea.